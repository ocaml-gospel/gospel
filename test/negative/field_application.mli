type t = { a: int }

val f : t -> int
(*@ r = f x
      ensures r = a x *)
