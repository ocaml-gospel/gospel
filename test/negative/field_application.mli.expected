
*******************************
********** Parsed file ********
*******************************
type t = {
  a: int }
val f : t -> int[@@gospel {| r = f x
      ensures r = a x |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Gospelstdlib *)

type t = {
a: int }
  

val f : t -> int
(*@ r = f x
    ensures ...
     *)
File "field_application.mli", line 5, characters 18-19:
Error: Record field a cannot be applied
